Session/OPM Changes

sarath-sasi-panickaveetil opened the pull request:
> Added Bar chart in OPM component
> Datalabels disabled in barcharts and added tooltip on clicking the bars
> Tooltip with primary and secondary data in session pages (when both primary and secondary datas shown)
> Minor alignment fixes (Calendar icon and filter input alignments)

sarath-sasi-panickaveetil committed:
rebasing with develop branch

sarath-sasi-panickaveetil committed:
Session/OPM Changes
> Added Bar chart in OPM page
> Datalabel disabled and tooltip on click
> Tooltip with primary and secondary data
> Minor alignment fixes

thasnim-cm commented on src/components/LineChart.tsx:
```diff
@@ -61,25 +67,41 @@ function LineChart({
         rotate
           ? "rotate-90 h-[390px] w-[90vh] bg-inherit ml-[-50vw] mt-[22vh]"
-          : "relative ml-[5vw] sm:ml-[1vw] mr-[5vw] sm:mr-[0] md:mr-[0] sm:h-[340px] bg-[#22262C]"
+          : "relative ml-[5vw] sm:ml-[1rem] mr-[5vw] sm:mr-[0] md:mr-[0] sm:h-[340px] bg-[#22262C]"
```
md:mr-[0] can be removed as sm and md values are same

sarath-sasi-panickaveetil replied:
done

thasnim-cm commented on src/views/opm.tsx:
```diff
@@ -328,16 +363,28 @@ const OPM: React.FC = () => {
-          />
+          <>
+            {tabValue === 0 ? (
+              <div className="home-opm border-0 rounded-[10px] w-full sm:w-[89vw] lg:w-full lg:ml-[0] h-[380px] lg:h-[380px] lg:mt-[3vh] top-[-5vh] relative ml-[5vw] sm:ml-[1vw] mr-[5vw] sm:mr-[0] md:mr-[0] sm:h-[340px] bg-[#22262C]">
```
md:mr-[0] can be removed as sm and md values are same

sarath-sasi-panickaveetil replied:
done

thasnim-cm commented on src/views/opm.tsx:
```diff
@@ -328,16 +363,28 @@ const OPM: React.FC = () => {
+            ) : (
+              <LineChart
+                title={TITLE.OPM}
+                className="home-opm border-0 rounded-[10px] w-full sm:w-[89vw] lg:w-full lg:ml-[0] h-[380px] lg:h-[380px] lg:mt-[3vh] top-[-5vh]"
```
lg:h-[380px] can be removed

sarath-sasi-panickaveetil replied:
Cannot remove this since we have value against sm

raj-kumar-litmus commented on src/components/LineChart.tsx:
```diff
@@ -61,25 +67,41 @@ function LineChart({
           </div>
         )}
-      <Line options={options} data={data} />
+      {tabValue === 1 ? (
```
LineChart is supposed to be re-usable.
That is why I keep linechart.tsx separate from opm.tsx and opmcomparison.tsx;

But after these changes, it looks like there is no way to use Linechart.tsx without including a custom tab for Barchart.

Please find a workaround so that LineChart.tsx is just for Line charts and can be re-used.

sarath-sasi-panickaveetil replied:
Removed Barchart from the component and created a new Barchart comp

raj-kumar-litmus commented on src/config/chartConfig.tsx:
```diff
@@ -88,3 +94,64 @@ export const BAR_CHART_OPTIONS: Chart.ChartOptions = {
   },
 };
+
+export const OPM_BAR_CHART_OPTIONS: Chart.ChartOptions = {
```
Move this to `appconstants`; That is where we have kept other chart options.
And see if you can use spread operator to avoid duplicate lines of code.

sarath-sasi-panickaveetil replied:
seems like there are config-related changes in Adithya's PR. I will move all the config to config.js once all those changes are in place to avoid conflicts. Used spread operator in possible places

sarath-sasi-panickaveetil committed:
addressed review comments

sarath-sasi-panickaveetil committed:
addressed review comments

thasnim-cm committed:
[fix] Quick links

sarath-sasi-panickaveetil committed:
rebase develop branch
> added opm barchart minor css fix in home page

sarath-sasi-panickaveetil committed:
Demo review by Rahul

> Timezone issue (X axis labelling) in OPM Comparison and OPM charts
> Datalabels in OPM Bar Charts
> Show Y axis in all bar charts(Home, Session and OPM pages
> Datalabels added for 10 mins data in session page.
> Datalabels added for 15 mins data in OPM page bar charts.
> Bar chart bar width increased and reduce the space between bars

sarath-sasi-panickaveetil committed:
App constants and config changes

sarath-sasi-panickaveetil commented:
Closing the PR due to the added files after the merge. 
New PR with same changes
 https://github.com/raj-kumar-litmus/support-hub/pull/41/
